### ISSUE TITLE
[ProgressIndicator] Not updated inside recyclerview on scrolling

### ISSUE THREAD

**Issue by ahetawal-p** [material-components/material-components-android#1681]
I am using `ProgressIndicator` inside a recycler view with each row having its own `ProgressIndicator`.
Some rows hide the indicator based on some business logic.
Now if I scroll or pull to refresh on my recyler view, sometimes the indicator is hidden even for valid rows. 
I feel the visibility state for indicator is being shared somehow between various rows, which is causing this bug.

```
 <com.google.android.material.progressindicator.ProgressIndicator
            android:id="@+id/progress"
            style="@style/Widget.MaterialComponents.ProgressIndicator.Linear.Determinate"
            android:layout_width="70dp"
            android:layout_height="12dp"
            app:layout_constraintBottom_toBottomOf="parent"
            app:layout_constraintEnd_toEndOf="parent"
            app:layout_constraintTop_toTopOf="parent"
            tools:progress="67" />


it.progress.completedPercent == 0F -> {
                    progress.isGone = true
                }
                else -> {
                    progress.setProgressCompat(it.progress.completedPercent.toInt(), false)
                    progress.isGone = false
                }
```

Material Android Library version `1.3.0-alpha02`

**Device:** Android Emulator: Api 29

**Comment by bibi136** [comment]
I got the same, added `progress.show()` on the Adapter 's `onBindViewHolder()` to temporary fix this issue.

**Comment by pekingme** [comment]
Obsolete, as `ProgressIndicator` has been divided into `LinearProgressIndicator` and `CircularProgressIndicator`. Similar bugs are reported and fixed in #1914 and #1902.
